File: webservers/php/webserver.php
<?php
$uri = $_SERVER['REQUEST_URI'];
$method = $_SERVER['REQUEST_METHOD'];

try {
    if ($method === 'POST') {
        $input = $_POST['input'] ?? throw new Exception('Input is required');
        echo match ($uri) {
            '/sha256' => hash('sha256', $input),
            '/base64' => base64_encode($input),
            '/urlencode' => urlencode($input),
            default => http_response_code(404) . 'Not Found',
        };
    } elseif ($method === 'GET') {
        http_response_code(201);
        echo 'php';
    } else {
        http_response_code(405);
        echo 'Method Not Allowed';
    }
} catch (Exception $e) {
    http_response_code(500);
    echo $e->getMessage() ?? 'Internal Server Error';
}

File: webservers/python/webserver.py
from flask import Flask, request, make_response, abort
import hashlib
import base64
import urllib.parse

app = Flask(__name__)

@app.route('/', defaults={'path': ''}, methods=['GET', 'POST'])
@app.route('/<path:path>', methods=['GET', 'POST'])
def handle_request(path):
    try:
        method = request.method
        uri = '/' + path

        if method == 'POST':
            input_value = request.form.get('input')
            if input_value is None:
                raise Exception('Input is required')

            if uri == '/sha256':
                output = hashlib.sha256(input_value.encode()).hexdigest()
                return output
            elif uri == '/base64':
                output = base64.b64encode(input_value.encode()).decode()
                return output
            elif uri == '/urlencode':
                output = urllib.parse.quote(input_value)
                return output
            else:
                return 'Not Found', 404

        elif method == 'GET':
            response = make_response('python', 201)
            return response

        else:
            return 'Method Not Allowed', 405

    except Exception as e:
        return str(e) or 'Internal Server Error', 500

if __name__ == '__main__':
    app.run(port=8008)
